Troupon.Catalog.Api/DependencyInjectionExtensions/SwaggerHelper.cs

```diff
@@ -0,0 +1,179 @@
+
+          return true;
+        });
+
+        Auth2FiltersAndSecurity(apiKeySettings, setup);
+
+        setup.EnableAnnotations();
+        setup.IgnoreObsoleteActions();
+        setup.OrderActionsBy(descriptor => descriptor.GroupName);
+        setup.ResolveConflictingActions(api => api.First());
+      });
+      services.Configure<ApiBehaviorOptions>(
+            options =>
+            {
+              options.InvalidModelStateResponseFactory = actionContext =>
+              {
+                var actionExecutingContext =
+                  actionContext as ActionExecutingContext;
+
+                if (actionContext.ModelState.ErrorCount > 0
+                    && actionExecutingContext?.ActionArguments.Count ==
+                    actionContext.ActionDescriptor.Parameters.Count)
+                {
+                  return new UnprocessableEntityObjectResult(actionContext.ModelState);
+                }
+
+                return new BadRequestObjectResult(actionContext.ModelState);
+              };
+            });
+
+      return services;
+    }
+
+    private static void Auth2FiltersAndSecurity(IOAuthSettings apiKeySettings, SwaggerGenOptions setup)
+    {
+      setup.SchemaFilter<SchemaFilter>();
+
+      setup.MapType<FileContentResult>(() => new OpenApiSchema { Type = "string", Format = "binary" });
+      setup.MapType<IFormFile>(() => new OpenApiSchema { Type = "string", Format = "binary" });
+      setup.AddBearerAuthentication(apiKeySettings);
+    }
+
+    private static void ConfigureSwaggerGenPerVersion(SwaggerGenOptions setup, ApiVersionDescription description)
+    {
+      setup.SwaggerDoc(description.GroupName, new OpenApiInfo
+      {
+        Title = $"Troupon.Catalog Api Specification {description.ApiVersion.ToString()}",
+        Description = "Api specification",
+        Version = description.ApiVersion.ToString(),
+        Contact = new OpenApiContact()
+        {
+          Email = "oualid.ktata@gmail.com",
+          Name = "Oualid Ktata",
+        },
+        License = new OpenApiLicense()
+        {
+          Name = "OKT",
+          Url = new Uri("https://opensource.org/licenses/MIT"),
+        },
+      });
+    }
+
+    public static void ConfigureSwaggerUI(this IApplicationBuilder app, IApiVersionDescriptionProvider apiVersionDescriptionProvider, IOAuthSettings authSettings)
+    {
+      app.UseSwaggerUI(setup =>
+      {
+        foreach (var description in apiVersionDescriptionProvider.ApiVersionDescriptions)
+        {
+          setup.SwaggerEndpoint($"/swagger/{description.ApiVersion}/swagger.json", $"Troupon Catalog Specification{description.ApiVersion.ToString()}");
+          setup.RoutePrefix = string.Empty;
+        }
+
+        setup.RoutePrefix = string.Empty;
+        AddOAuth2(authSettings, setup);
+      });
+    }
+
+    private static void AddOAuth2(IOAuthSettings authSettings, SwaggerUIOptions setup)
+    {
+      // setup.OAuthUseBasicAuthenticationWithAccessCodeGrant();
```

Comment:
Please remove commented out code. If we need to bring it back, it's in the git repository.

Reply:
Done!